src/dmd/traits.d

```diff
@@ -2200,3 +2231,128 @@ Lnext:
     }
     return true;
 }
+
+/// compare arguments of __traits(isSame2)
+private bool isSame2(RootObject o1, RootObject o2, Scope* sc)
```

Comment:
This function doesn't seem to be called anywhere.

Reply:
Fixed.

Reply:
There seems to be a lot of overlap between these private `isSame`  and `isSame2` functions. Can you maybe get rid of the duplication?